orion/pkg/repos/semester_repo_test.go

```diff
@@ -173,3 +205,55 @@ func TestDeleteSemester(t *testing.T) {
 		t.Errorf("Unfulfilled expectations: %s", err)
 	}
 }
+
+//
+// Publish
+//
+func TestPublishSemesters(t *testing.T) {
+	db, mock, repo := initSemesterTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	result := sqlmock.NewResult(1, 1)
+	mock.ExpectPrepare(`^UPDATE semesters SET published_at=\? WHERE semester_id=\? AND published_at IS NULL`).
+		ExpectExec().
+		WithArgs(sqlmock.AnyArg(), "2020_fall").
+		WillReturnResult(result)
+	err := repo.Publish("2020_fall")
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+
+	// Validate results
+	if err := mock.ExpectationsWereMet(); err != nil {
+		t.Errorf("Unfulfilled expectations: %s", err)
+	}
+}
+
+//
+// Helper Methods
+//
+func getSemesterRows() *sqlmock.Rows {
+	return sqlmock.NewRows([]string{"Id", "CreatedAt", "UpdatedAt", "DeletedAt", "PublishedAt", "SemesterId", "Title"}).
+		AddRow(
+			1,
+			now,
+			now,
+			domains.NewNullTime(time.Time{}),
+			domains.NewNullTime(time.Time{}),
+			"2020_fall",
+			"Fall 2020",
+		)
+}
+
+func getSemester() domains.Semester {
+	return domains.Semester{
+		Id:          1,
+		CreatedAt:   now,
+		UpdatedAt:   now,
+		DeletedAt:   domains.NewNullTime(time.Time{}),
+		PublishedAt: domains.NewNullTime(time.Time{}),
+		SemesterId:  "2020_fall",
+		Title:       "Fall 2020",
+	}
+}
```

Comment:
I guess I can kinda see why you separate these into separate methods, since it's done 5 or 6 times. I personally don't feel the code's long enough to warrant that, but it's not a big deal.